src/main/java/org/truffleruby/parser/ast/HashParseNode.java

```diff
@@ -44,6 +44,9 @@
 public class HashParseNode extends ParseNode implements ILiteralNode {
     private final List<ParseNodeTuple> pairs;
 
+    /** Does this hash parse node represent formal keyword arguments? */
```

Comment:
What does formal mean here?
I would just drop the word, or replace it by `literal` but that might be confusing for `**h` and people might think that only means `foo: value`.
I'd say both are keyword arguments, simply.

Reply:
It's for the case of `foo(a: x, b: y)` and not the case of `foo({a: x, b: y})` - currently the parser cannot distinguish.

Reply:
The former is keyword arguments, the second isn't keyword arguments, it's just a positional Hash argument.

Reply:
So just 'represent keyword arguments'?

Reply:
yes